actions/plugins/version-bump-changelog/action.yml

```diff
@@ -0,0 +1,110 @@
+name: Bump npm version and release
+description: |
+  Bump npm version, create a git tag and optionally generate a changelog on main.
+
+inputs:
+  version:
+    description: "Semver type of new version (major / minor / patch)"
+    required: true
+    type: choice
+    options:
+      - patch
+      - minor
+      - major
+  generate-changelog:
+    description: "Generate changelog"
+    required: false
+    type: boolean
+    default: true
+
+runs:
+  using: composite
+  steps:
+    - name: Get secrets from Vault
+      id: get-secrets
+      uses: grafana/shared-workflows/actions/get-vault-secrets@main # zizmor: ignore[unpinned-uses]
+      env:
+        VAULT_INSTANCE: ops
+      with:
+        vault_instance: ${{ env.VAULT_INSTANCE }}
+        common_secrets: |
+          GITHUB_APP_ID=plugins-platform-bot-app:app-id
+          GITHUB_APP_PRIVATE_KEY=plugins-platform-bot-app:private-key
+
+    - name: Generate GitHub token
+      id: generate-github-token
+      uses: actions/create-github-app-token@3ff1caaa28b64c9cc276ce0a02e2ff584f3900c5 # v2.0.2
+      with:
+        app-id: ${{ env.GITHUB_APP_ID }}
+        private-key: ${{ env.GITHUB_APP_PRIVATE_KEY }}
+        owner: ${{ github.repository_owner }}
+
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      with:
+        token: ${{ steps.generate-github-token.outputs.token }}
+        persist-credentials: true
+        fetch-depth: 0
+
+    - name: Setup Git
+      shell: bash
+      run: |
+        git config user.name 'grafana-plugins-platform-bot[bot]'
+        git config user.email '144369747+grafana-plugins-platform-bot[bot]@users.noreply.github.com'
+
+    - name: Get previous tag before bumping
+      id: previous-tag
+      if: ${{ inputs.generate-changelog == 'true' }}
+      shell: bash
+      run: |
+        PREVIOUS_TAG=$(git describe --tags --abbrev=0 2>/dev/null || echo "v0.0.0")
+        echo "previous-tag=${PREVIOUS_TAG}" >> $GITHUB_OUTPUT
+
+    - name: Bump version
+      id: bump
+      shell: bash
+      run: |
+        NEW_VERSION=$(npm version ${INPUT_VERSION})
+        echo "new-version=${NEW_VERSION}" >> $GITHUB_OUTPUT
+      env:
+        INPUT_VERSION: ${{ inputs.version }}
+        GITHUB_TOKEN: ${{ steps.generate-github-token.outputs.token }}
+    
+    - name: Generate changelog
+      if: ${{ inputs.generate-changelog == 'true' }}
+      shell: bash
+      run: |
+        npm install -g generate-changelog
```

Comment:
nit: instead of installing this globally can you call `npx generate-changelog` in the places you need it.